lib/TokenManager.ts

```diff
@@ -46,23 +52,42 @@ function hasExpired(tokenMgmtRef, token) {
 }
 
 function emitExpired(tokenMgmtRef, key, token) {
-  tokenMgmtRef.emitter.emit('expired', key, token);
+  tokenMgmtRef.emitter.emit(EVENT_EXPIRED, key, token);
 }
 
 function emitRenewed(tokenMgmtRef, key, freshToken, oldToken) {
-  tokenMgmtRef.emitter.emit('renewed', key, freshToken, oldToken);
+  tokenMgmtRef.emitter.emit(EVENT_RENEWED, key, freshToken, oldToken);
 }
 
 function emitAdded(tokenMgmtRef, key, token) {
-  tokenMgmtRef.emitter.emit('added', key, token);
+  tokenMgmtRef.emitter.emit(EVENT_ADDED, key, token);
 }
 
-function emitRemoved(tokenMgmtRef, key, token) {
-  tokenMgmtRef.emitter.emit('removed', key, token);
+function emitRemoved(tokenMgmtRef, key, token?) {
+  tokenMgmtRef.emitter.emit(EVENT_REMOVED, key, token);
 }
 
 function emitError(tokenMgmtRef, error) {
-  tokenMgmtRef.emitter.emit('error', error);
+  tokenMgmtRef.emitter.emit(EVENT_ERROR, error);
+}
+
+function emitEventsForCrossTabsStorageUpdate(tokenMgmtRef, newValue, oldValue) {
+  const oldTokens = getTokensFromStorageValue(oldValue);
+  const newTokens = getTokensFromStorageValue(newValue);
+  Object.keys(newTokens).forEach(key => {
+    const oldToken = oldTokens[key];
+    const newToken = newTokens[key];
+    if (JSON.stringify(oldToken) !== JSON.stringify(newTokens)) {
```

Comment:
did you mean `newToken` (instead of `newTokens`) ?

Reply:
Thanks for catching it, dammit auto-completion...

Reply:
That's why I don't use auto-completion, and make boutique, artisanal typos...

Reply:
That said, if the above didn't trigger a test failure, that signals a test we should add...